testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js

```diff
@@ -1280,6 +1280,53 @@ QUnit.module('popup', moduleConfig, () => {
             assert.strictEqual(exception, null);
         }
     });
+
+    QUnit.test('widget has a correct popup height for the first opening if the pageSize equal dataSource length (T942881)', function(assert) {
+        const items = [{
+            id: 1,
+            value: 'value11'
+        }, {
+            id: 2,
+            value: 'value12'
+        }];
+        const onContentReadySpy = sinon.spy();
+        const dropDownList = $('#dropDownList').dxDropDownList({
+            displayExpr: 'value',
+            valueExpr: 'id',
+            dataSource: new DataSource({
+                store: [],
+                paginate: true,
+                pageSize: 2
+            }),
+            deferRendering: false,
+            opened: true,
+            onContentReady: onContentReadySpy
+        }).dxDropDownList('instance');
+        const listInstance = $(`.${LIST_CLASS}`).dxList('instance');
+        listInstance.option('pageLoadMode', 'scrollBottom');
+        listInstance.option('useNativeScrolling', 'true');
+        dropDownList.close();
+
+        const dataSource = new DataSource({
+            store: items,
+            paginate: true,
+            pageSize: 2
+        });
+        dropDownList.option('dataSource', dataSource);
```

Comment:
to my mind it's better to not define separate variable for `dataSource`, because we use it only once